src/Raven.Server/Documents/Subscriptions/LiveSubscriptionPerformanceCollector.cs

```diff
@@ -0,0 +1,304 @@
+            {
+                Stats.Enqueue(initialStats);
+            }
+            
+            Start();
+        }
+
+        // Upon collector init only: get history stats + in progress stats (to be sent to ws)
+        protected IEnumerable<SubscriptionTaskPerformanceStats> PrepareInitialPerformanceStats()
+        {
+            var results = new List<SubscriptionTaskPerformanceStats>();
+            
+            using (Database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var allSubscriptions = Database.SubscriptionStorage.GetAllSubscriptions(context, false, 0, int.MaxValue);
+                foreach (var subscription in allSubscriptions)
+                {
+                    var subscriptionName = subscription.SubscriptionName;
+                    var subscriptionId = subscription.SubscriptionId;
+                    
+                    results.Add(new SubscriptionTaskPerformanceStats
+                    {
+                        TaskName = subscriptionName,
+                        TaskId = subscriptionId
+                    });
+                    
+                    _perSubscriptionConnectionStats[subscriptionName] = new SubscriptionAndPerformanceConnectionStatsList(subscription);
+                    _perSubscriptionBatchStats[subscriptionName] = new SubscriptionAndPerformanceBatchStatsList(subscription);
+                }
+
+                foreach (var kvp in _perSubscriptionConnectionStats)
+                {
+                    var connectionPerformance = new List<SubscriptionConnectionPerformanceStats>();
+                    
+                    var connectionState = Database.SubscriptionStorage.GetSubscriptionConnection(context, kvp.Value.Handler.SubscriptionName);
+                    if (connectionState != null)
+                    {
+                        var currentConnection = connectionState.Connection;
+                        var connectionAggregators = new List<SubscriptionConnectionStatsAggregator>();
+
+                        if (currentConnection != null)
+                        {
+                            // add inProgress aggregator
+                            connectionAggregators.Add(currentConnection.GetPerformanceStats()); 
+                        }
+
+                        // add history aggregators
+                        connectionAggregators.AddRange(connectionState.RecentConnections.Select(x => x.GetPerformanceStats()));
+                        connectionAggregators.AddRange(connectionState.RecentRejectedConnections.Select(x => x.GetPerformanceStats()));
+
+                        // add connection stats to results 
+                        connectionPerformance.AddRange(connectionAggregators.Select(x => x.ToConnectionPerformanceLiveStatsWithDetails()));
+                        var subscriptionItem = results.Find(x => x.TaskName == kvp.Key);
```

Comment:
can we match by some numeric id?

Reply:
in batches we have if https://github.com/ravendb/ravendb/pull/11766/files#diff-875e3ec22a8b47240e4a97b0ec9e862136bc6243585a7d2978b75c637359d15fR119 here we don't have it. Is it ok?

Reply:
done